chat/src/agent/tools.py

```diff
@@ -63,3 +64,25 @@ def aggregate(agg_field: str, term_field: str, term: str):
         return json.dumps(response, default=str), response
     except Exception as e:
         return json.dumps({"error": str(e)}), None
+
+@tool(response_format="content_and_artifact")
+def retrieve_documents(doc_ids: List[str]):
+    """
+    Retrieve documents from the OpenSearch index based on a list of document IDs. 
+
+    Use this instead of the search tool if the user has provided docs for context
+    and you need the full metadata in order to answer their question.
+
+    Args:
+        doc_ids (List[str]): A list of document IDs to fetch.
+
+    Returns:
+        A JSON list of documents that match the given IDs.
+    """
+    
+    try:
+        response = opensearch_vector_store().retrieve_documents(doc_ids)
+        documents = [hit["_source"] for hit in response["hits"]["hits"]]
```

Comment:
This list comprehension was throwing an error. I adjust it to this:

```python
documents = []
        for doc in response:
            metadata = doc.metadata
            if 'embedding' in metadata:
                metadata.pop('embedding')
            documents.append(metadata)
```

I figured there was no need to include the `embedding` field, but maybe…